record_imu_cam/plugin.cpp

```diff
@@ -0,0 +1,101 @@
+#include "common/plugin.hpp"
+#include "common/switchboard.hpp"
+#include "common/data_format.hpp"
+
+#include <fstream>
+#include <boost/filesystem.hpp>
+#include <iomanip>
+#include <opencv2/core/mat.hpp>
+#include <opencv2/imgcodecs.hpp>
+#include <opencv2/imgproc.hpp>
+
+using namespace ILLIXR;
+/*
+The plugin to record the IMU and the camera data. It is written by Ying Jing and Henry Che from Illinois.
+*/
+
+class record_imu_cam : public plugin {
+public:
+	record_imu_cam(std::string name_, phonebook* pb_)
+		: plugin{name_, pb_}
+		, sb{pb->lookup_impl<switchboard>()}
+	{
+		boost::filesystem::path record_data = get_record_data_path();
+		//check folder exist, if exist delete it'
+		boost::filesystem::remove_all(record_data);
+
+		boost::filesystem::path imu_dir = record_data / "imu0";
+		boost::filesystem::create_directories(imu_dir);
+		std::string imu_file = imu_dir.string() + "/data.csv";
+		imu_wt_file.open(imu_file, std::ofstream::out);
+		imu_wt_file << "#timestamp [ns],w_x [rad s^-1],w_y [rad s^-1],w_z [rad s^-1],a_x [m s^-2],a_y [m s^-2],a_z [m s^-2]" << std::endl;
+
+		boost::filesystem::path cam0_dir = record_data / "cam0";
+		boost::filesystem::create_directories(cam0_dir);
+		std::string cam0_file = cam0_dir.string() + "/data.csv";
+		cam0_wt_file.open(cam0_file, std::ofstream::out);
+		cam0_wt_file << "#timestamp [ns],filename" << std::endl;
+
+		boost::filesystem::path cam1_dir = record_data / "cam1";
+		boost::filesystem::create_directories(cam1_dir);
+		std::string cam1_file = cam1_dir.string() + "/data.csv";
+		cam1_wt_file.open(cam1_file, std::ofstream::out);
+		cam1_wt_file << "#timestamp [ns],filename" << std::endl;
+
+		sb->schedule<imu_cam_type>(id, "imu_cam", [this](switchboard::ptr<const imu_cam_type> datum, std::size_t){
+			this->dump_data(datum);
+		});
+	}
+
+	void dump_data(switchboard::ptr<const imu_cam_type> datum) {
+		boost::filesystem::path record_data = get_record_data_path();
```

Comment:
Instead of calling this function every time, store the result in a class member in the constructor and then use it instead.